File: src/main/java/org/noxet/noxetserver/playerstate/properties/PSPCollidable.java
package org.noxet.noxetserver.playerstate.properties;

import org.bukkit.entity.Player;
import org.noxet.noxetserver.playerstate.PlayerStateProperty;

public class PSPCollidable implements PlayerStateProperty<Boolean> {
    @Override
    public String getConfigName() {
        return "collidable";
    }

    @Override
    public Boolean getDefaultSerializedProperty() {
        return true;
    }

    @Override
    public Boolean getSerializedPropertyFromPlayer(Player player) {
        return player.isCollidable();
    }

    @Override
    public void restoreProperty(Player player, Boolean collidable) {
        player.setCollidable(collidable);
    }
}

File: src/main/java/org/noxet/noxetserver/playerstate/properties/PSPCollidable.scala
package org.noxet.noxetserver.playerstate.properties

import org.bukkit.entity.Player
import org.noxet.noxetserver.playerstate.PlayerStateProperty

object PSPCollidable extends PlayerStateProperty[Boolean]:
  override def getConfigName: String = "collidable"

  override def getDefaultSerializedProperty: Boolean = true

  override def getSerializedPropertyFromPlayer(player: Player): Boolean =
    player.isCollidable

  override def restoreProperty(player: Player, collidable: Boolean): Unit =
    player.setCollidable(collidable)